### PR TITLE
R tinn/student t likelihood

### PR DESCRIPTION
**Pull request by r-tinn**

### CODE REVIEW

File: src/Tutorials/GausianProcessRegressor.cs [hunk 1]
```diff
@@ -39,35 +40,51 @@ public void Run()
...
+                GaussianProcess gp = new GaussianProcess(new ConstantFunction(0), kf);
+
+                // Convert SparseGP to full Gaussian Process by evaluating at all the training points
+                prior.ObservedValue = new SparseGP(new SparseGPFixed(gp, trainingInputs.ToArray()));
+                double logOdds = engine.Infer<Bernoulli>(evidence).LogOdds;
+                Console.WriteLine("{0} evidence = {1}", kf, logOdds.ToString("g4"));
+
+                // Infer the posterior Sparse GP
+                SparseGP sgp = engine.Infer<SparseGP>(f);
+                //PlotPredictions(sgp, trainingInputs, trainingOutputs, likelihoodDescription);
```
Comment (philipjball) [inline src/Tutorials/GausianProcessRegressor.cs]: Is it worth having some sort of console printout of statistics (i.e., RMSE) in the case the user can't use OxyPlot at all?

File: src/Tutorials/GausianProcessRegressor.cs [hunk 2]
```diff
@@ -102,13 +119,33 @@ private void PlotPredictions(SparseGP sgp, Vector[] trainingInputs, double[] tra
...
+        #endif
+
+        private Variable<double> GetScore(VariableArray<Vector> x, Variable<IFunction> f, Range j)
+        {
+            return Variable.FunctionEvaluate(f, x[j]);
+        }
+
+        private VariableArray<double> GetNoisyScore(VariableArray<Vector> x, Variable<IFunction> f, Range j, double[] trainingOutputs)
+        {
+            // The stduent-t distribution arises as the mean of a normal distribution once an unknown precision is marginalised out
```
Comment (philipjball) [inline src/Tutorials/GausianProcessRegressor.cs]: Typo: student

File: src/Tutorials/GausianProcessRegressor.cs [hunk 3]
```diff
@@ -39,35 +40,51 @@ public void Run()
...
+                // Infer the posterior Sparse GP
+                SparseGP sgp = engine.Infer<SparseGP>(f);
+                //PlotPredictions(sgp, trainingInputs, trainingOutputs, likelihoodDescription);
+            }
         }
-#if oxyplot
-        private void PlotPredictions(SparseGP sgp, Vector[] trainingInputs, double[] trainingOutputs)
+
+        #if oxyplot
+        private void PlotPredictions(SparseGP sgp, Vector[] trainingInputs, double[] trainingOutputs, string likelihoodDescription)
```
Comment (philipjball) [inline src/Tutorials/GausianProcessRegressor.cs]: NIT: Reusing the `likelihoodDescription` variable will give some ugly file names; maybe have a conditional and just use `GPRegressionPredictionsStudentT.png` instead?

**Review by philipjball — CHANGES_REQUESTED** [review]
Seems good, just a few minor points.

File: src/Tutorials/GausianProcessRegressor.cs (hunk 1 above)
Comment (r-tinn) [reply src/Tutorials/GausianProcessRegressor.cs]: Ah yes I remember you mentioning this before. Did you say you had a function for RMSE? I can't seem to find it

Comment (philipjball) [reply src/Tutorials/GausianProcessRegressor.cs]: Also maybe worth including the preprocessor directive for the Plot line. I'll hunt for the RMSE code.

Comment (r-tinn) [reply src/Tutorials/GausianProcessRegressor.cs]: Thanks mate!

Comment (philipjball) [reply src/Tutorials/GausianProcessRegressor.cs]:
```
            double sqDiff = 0;
            for (int i = 0; i < outputs.Length; i++)
            {
                Gaussian post = sgp.Marginal(inputs[i]);
                double postMean = post.GetMean();
                sqDiff += System.Math.Pow(postMean - outputs[i], 2);
                Console.WriteLine("f({0}) = {1} (actual {2})", inputs[i], post, outputs[i]);
            }
            double rmse = System.Math.Sqrt(sqDiff / outputs.Length);
            Console.WriteLine("RMSE is: {0}", rmse);

Comment (r-tinn) [reply src/Tutorials/GausianProcessRegressor.cs]: nice

**Pull request merged**